autofs-5.0.5 - dont check null cache on expire

From: Ian Kent <raven@themaw.net>

When expiring an entry there is no need to check the null map
entry cache. If we have a mount then it must have been done at
some point when the entry was not a nulled so it still needs
to be expired. Remove this check.
---

 CHANGELOG          |    1 +
 daemon/automount.c |    9 ---------
 2 files changed, 1 insertion(+), 9 deletions(-)


--- autofs-5.0.5.orig/CHANGELOG
+++ autofs-5.0.5/CHANGELOG
@@ -36,6 +36,7 @@
 - fix remount locking.
 - fix wildcard map entry match.
 - fix parse_sun() module init.
+- dont check null cache on expire.
 
 03/09/2009 autofs-5.0.5
 -----------------------
--- autofs-5.0.5.orig/daemon/automount.c
+++ autofs-5.0.5/daemon/automount.c
@@ -526,20 +526,11 @@ static int umount_subtree_mounts(struct 
    it also tries to umount path itself */
 int umount_multi(struct autofs_point *ap, const char *path, int incl)
 {
-	struct mapent_cache *nc;
 	int is_autofs_fs;
 	int left;
 
 	debug(ap->logopt, "path %s incl %d", path, incl);
 
-	nc = ap->entry->master->nc;
-	cache_readlock(nc);
-	if (cache_lookup_distinct(nc, path)) {
-		cache_unlock(nc);
-		return 0;
-	}
-	cache_unlock(nc);
-
 	is_autofs_fs = 0;
 	if (master_find_submount(ap, path))
 		is_autofs_fs = 1;
